### PR TITLE
📖 docs: update CAPI support and guarantees

### PR DESCRIPTION
**Pull request by oscr**
**What this PR does / why we need it**:

Update support and guarantees

cc @kubernetes-sigs/cluster-api-release-team

### CODE REVIEW

**Review by killianmuldoon — APPROVED** [review]
/lgtm
/hold

Until the release date passes so we don't have to update :sweat_smile:

**Comment by sbueringer** [comment]
/lgtm
/approve

Not sure I understand why we want to hold. I assume so we don't have to update in case we can't make the release date?
(but of course I'm fine with holding :))

**Review by furkatgofurov7 — COMMENTED** [review]
LGTM

File: CONTRIBUTING.md
Comment (fabriziopandini) [inline CONTRIBUTING.md:158]:
```suggestion
| v1.3.x        | **v1beta1**  | when v1.5.0 will be released, tentatively July 2023                |
```

**Review by fabriziopandini — COMMENTED** [review]
Thanks for preparing this PR
Also, what about adding EOL on all the mentions to 1.2 in https://cluster-api.sigs.k8s.io/reference/versions.html; ideally we should also drop 1.1 columns and add 1.5 columns, but I'll defer to you if to do this in one PR or to split the work

File: CONTRIBUTING.md
Comment (oscr) [reply CONTRIBUTING.md]: No problem and we got plenty of time. Added and updated.

**Pull request merged**
